src/uct/tcp/tcp_ep.c

```diff
@@ -1452,6 +1452,14 @@ uct_tcp_ep_am_prepare(uct_tcp_iface_t *iface, uct_tcp_ep_t *ep,
     *hdr          = ep->tx.buf;
     (*hdr)->am_id = am_id;
 
+    ++ep->tx.sn;
+    if (ep->tx.sn == ep->last_acked_sn) {
```

Comment:
how can it be?

Reply:
e.g. if we don't request ACK for long period of time. so, it happens that `ep->tx.sn == ep->last_acked_sn == 2^32 - 1`

Reply:
do you mean that `ep->tx.sn` will wrap? Then why they will be necessarily `2^32 - 1`

Reply:
Yes, the value can wrap.
No, it could be another value, just an example, if no PUT operations done at all

Reply:
I have an idea - will prepare a commit